python/pyiceberg/table/sorting.py

```diff
@@ -0,0 +1,133 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+# pylint: disable=keyword-arg-before-vararg
+
+from enum import Enum
+from typing import (
+    Any,
+    Callable,
+    Dict,
+    List,
+    Optional,
+    Union,
+)
+
+from pydantic import Field, root_validator
+
+from pyiceberg.schema import Schema
+from pyiceberg.transforms import Transform, UnboundTransform
+from pyiceberg.types import IcebergType
+from pyiceberg.utils.iceberg_base_model import IcebergBaseModel
+
+
+class SortDirection(Enum):
+    ASC = "asc"
+    DESC = "desc"
+
+
+class NullOrder(Enum):
+    NULLS_FIRST = "nulls-first"
+    NULLS_LAST = "nulls-last"
+
+
+class SortField(IcebergBaseModel):
+    """A field that provides information on how the table is sorted
+
+    Args:
+      source_id (int): Source column id from the table’s schema
+      transform (str): Transform that is used to produce values to be sorted on from the source column.
+                       This is the same transform as described in partition transforms.
+      source_id (SortDirection): Sort direction, that can only be either asc or desc
+      source_id (NullOrder): Null order that describes the order of null values when sorted. Can only be either nulls-first or nulls-last
+    """
+
+    class Config:
+        frozen = False
+
+    def __init__(
+        self,
+        source_id: Optional[int] = None,
+        transform: Optional[Union[Transform, Callable[[IcebergType], Transform]]] = None,
+        direction: Optional[SortDirection] = None,
+        null_order: Optional[NullOrder] = None,
+        **data: Any,
+    ):
+        if source_id is not None:
+            data["source-id"] = source_id
+        if transform is not None:
+            data["transform"] = transform
+        if direction is not None:
+            data["direction"] = direction
+        if null_order is not None:
+            data["null-order"] = null_order
+        super().__init__(**data)
+
+    @root_validator(pre=True)
+    def set_null_order(cls, values: Dict[str, Any]) -> Dict[str, Any]:
+        values["direction"] = values["direction"] if values.get("direction") else SortDirection.ASC
```

Comment:
This doesn't need to produce a `str` value? (Assuming since the type is `Any` it works, but this is unexpected for me)

Reply:
Of course, you can put everything in there. This will be checked when the validator does its job, but that hasn't been done at the `pre_validator` step.